src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -457,8 +505,8 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         path_str: &'path str,
         ns: Namespace,
         module_id: DefId,
-        extra_fragment: &Option<String>,
-    ) -> Result<(Res, Option<String>), ErrorKind<'path>> {
+        extra_fragment: &Option<UrlFragment>,
```

Comment:
I believe this and other instances of `extra_fragment` are always user-written fragments.